beacon-chain/db/kv/blob.go

```diff
@@ -80,61 +75,95 @@ func (s *Store) SaveBlobSidecar(ctx context.Context, scs []*ethpb.BlobSidecar) e
 			if slots.ToEpoch(scs[0].Slot) >= oldEpoch.Add(uint64(params.BeaconNetworkConfig().MinEpochsForBlobsSidecarsRequest)) {
 				if err := bkt.Delete(replacingKey); err != nil {
 					log.WithError(err).Warnf("Could not delete blob with key %#x", replacingKey)
+					return err
 				}
 			} else {
 				// Otherwise, we need to merge the new blob with the old blob.
-				enc := bkt.Get(replacingKey)
-				sc := &ethpb.BlobSidecars{}
-				if err := decode(ctx, enc, sc); err != nil {
-					return err
-				}
-				sc.Sidecars = append(sc.Sidecars, scs...)
-				sortSideCars(sc.Sidecars)
-				encodedBlobSidecar, err = encode(ctx, &ethpb.BlobSidecars{Sidecars: sc.Sidecars})
-				if err != nil {
+				existing = bkt.Get(replacingKey)
+				if err := decode(ctx, existing, sc); err != nil {
 					return err
 				}
 			}
 		}
-		return bkt.Put(newKey, encodedBlobSidecar)
+
+		sc.Sidecars = append(sc.Sidecars, scs...)
+		sortSideCars(sc.Sidecars)
+		var err error
```

Comment:
why do we need this var?

Reply:
because on the next line we assign to the struct field directly, so err needs to be defined in this scope.